DSTORE-643 Add unit tests for SkipExistingDocumentsProcessorFactory.

timatbw opened the pull request.

johnmbw commented on solr/core/src/test/org/apache/solr/update/processor/SkipExistingDocumentsProcessorFactoryTest.java:
```diff
@@ -0,0 +1,328 @@
+  // Tests for logic in the factory
+
+  @Test(expected=SolrException.class)
+  public void testExceptionIfSkipInsertParamNonBoolean() throws Exception {
```
I don't think you need the `throws Exception` here? Don't think there are any methods in the test that declare they throw a checked exception.

timatbw replied:
Thanks - will tidy up these things you've highlighted. I borrowed some of the test outline from another Solr test class, so the `throws` might be an artifact of that.

johnmbw replied:
Yeah I figured it was probably to match the solr code (was trying to restrain my pickiness a bit 😄 )

johnmbw commented on solr/core/src/test/org/apache/solr/update/processor/SkipExistingDocumentsProcessorFactoryTest.java:
```diff
@@ -0,0 +1,328 @@
+    doReturn(true).when(processor).doesDocumentExist(docId);
+
+    processor.processAdd(cmd);
+    verify(next, times(1)).processAdd(cmd);
```
NABD `times(1)` not needed here

johnmbw commented on solr/core/src/test/org/apache/solr/update/processor/SkipExistingDocumentsProcessorFactoryTest.java:
```diff
@@ -0,0 +1,328 @@
+    AddUpdateCommand cmd = new AddUpdateCommand(req);
+    cmd.setIndexedId(docId);
+    cmd.solrDoc = new SolrInputDocument();
+    cmd.solrDoc.addField("last_name", ImmutableMap.of("set", "Smith"));
```
Maybe worth putting this bit into a `private` method (`createAtomicUpdateCmd`)? As it'll read better in the tests and avoid a bit of duplication.

timatbw committed:
DSTORE-643 Add unit tests for SkipExistingDocumentsProcessorFactory

timatbw commented:
Have updated, removed many redundant `throws` clauses, refined others to `IOException`, removed redundant `times(1)`, refactored some repeated code, and added 2 new tests to check the leader logic.

johnmbw reviewed (approved):
LGTM
